### PR TITLE
Us8 edit

### PR DESCRIPTION
**Pull request by ekat-br**
This is PR for [US8](https://github.com/ekat-br/roomcheck/issues/14). Thanks for reviewing my code. You can view the results here https://roomcheck-cbtv6z6zv-ekat-br.vercel.app/.

### CODE REVIEW

File: pages/admin/[id].js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        console.error("Error fetching room data:", error);
+      }
+    }
+
+    if (id) {
+      fetchRoomData();
+    }
+  }, [id]);
+
+  console.log("roomData:", roomData);
```
Comment (Kheirah) [inline pages/admin/[id].js]: Remove `console.logs` before you create a PR. They're not supposed to stay in the code base forever =)

File: pages/admin/[id].js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import { useEffect } from "react";
```
Comment (Kheirah) [inline pages/admin/[id].js]:
```suggestion
import { useState, useEffect } from "react";
```
Shorter and more readable ;-)

File: components/RoomList/index.js [hunk 3]
```diff
@@ -1,37 +1,44 @@
...
+      setIsLoading(false);
+    } catch (error) {
+      console.error("Error fetching room data:", error);
+      setIsLoading(false);
     }
+  }
 
+  useEffect(() => {
     fetchRoomData();
   }, []);
```
Comment (Kheirah) [inline components/RoomList/index.js]: Why did you move the function outside the useEffect callback? Technically, when the function is _outside_, you need to pass it as a dependency into the dependency array. When it's inside, you don't. Better move it back into the callback.

File: components/RoomList/index.js [hunk 4]
```diff
@@ -1,37 +1,44 @@
...
-
   async function updateRoomState(roomId, newState) {
     //API Call for updating the room state
+    console.log(roomId);
```
Comment (Kheirah) [inline components/RoomList/index.js]: Removed unused console.logs :)

File: components/RoomButton/index.js [hunk 5]
```diff
@@ -32,6 +32,8 @@ export default function RoomButton({
       default:
         newState = "Keine Angabe";
     }
+    console.log("Toggling room state. Room ID:", room._id);
+    console.log("New State:", newState);
```
Comment (Kheirah) [inline components/RoomButton/index.js]: Remove unused logs =)

File: components/RoomEditForm/index.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+  useEffect(() => {
+    setFormData({
+      roomName: room?.name || "",
+      roomSubject: room?.subject || "",
+    });
+  }, [room]);
+
+  console.log("Room object:", room);
+  console.log("FormData:", formData);
```
Comment (Kheirah) [inline components/RoomEditForm/index.js]: Remove unused logs =)

File: components/RoomEditForm/index.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+  async function handleSubmitForm(event) {
+    event.preventDefault();
+
+    console.log("Request Payload:", {
+      roomName: formData.roomName,
+      roomSubject: formData.roomSubject,
+    });
```
Comment (Kheirah) [inline components/RoomEditForm/index.js]: Remove unused logs =)

File: components/RoomEditForm/index.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+export default function RoomEditForm({ room }) {
+  const router = useRouter();
+  const [formData, setFormData] = useState({});
+
+  useEffect(() => {
+    setFormData({
+      roomName: room?.name || "",
+      roomSubject: room?.subject || "",
+    });
+  }, [room]);
```
Comment (Kheirah) [inline components/RoomEditForm/index.js]: This does work, but you can make this an even easier by setting the initial state of your `formData`.
Instead of initializing it as an empty object you can pass the room prop as argument.
```suggestion
export default function RoomEditForm({ room }) {
  const router = useRouter();
  const [formData, setFormData] = useState({
      roomName: room?.name || "",
      roomSubject: room?.subject || "",
    });

```

Something like this.

**Review by Kheirah — CHANGES_REQUESTED** [review]
Very good!

Have a look at the comments and see if you can adjust your code.

File: components/RoomEditForm/index.js (hunk 8 above)
Comment (ekat-br) [reply components/RoomEditForm/index.js]: Thats because I need the default values when opening the edit form. With your suggested code I do not see default values.

**Comment by ekat-br** [comment]
Thanks for your feedback @Kheirah, I applied the suggested changes.

File: components/RoomEditForm/index.js (hunk 8 above)
Comment (Kheirah) [reply components/RoomEditForm/index.js]: Mh, I see. All right, if you can't make it work, leave it unchanged 👍🏼

**Review by Kheirah — APPROVED** [review]
Looking good!
